File: app/src/main/java/com/bkjcb/rqapplication/emergency/retrofit/EmergencyService.java
package com.bkjcb.rqapplication.emergency.retrofit;

import com.bkjcb.rqapplication.base.model.HttpResult;
import com.bkjcb.rqapplication.base.model.SimpleHttpResult;
import com.bkjcb.rqapplication.emergency.model.EmergencyItem;
import com.bkjcb.rqapplication.emergency.model.EmergencyModel;

import java.util.List;

import io.reactivex.Observable;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.POST;

/**
 * Created by DengShuai on 2019/10/31.
 * Description :
 */
public interface EmergencyService {
    @POST("/rq/push/SaveSiteInfo")
    @FormUrlEncoded
    Observable<HttpResult> submit(
            @Field("userId") String userId,
            @Field("shiguid") String uuid,
            @Field("reportingUnit") String reportingUnit,
            @Field("qushu") String qushu,
            @Field("remark") String remark,
            @Field("reportingDate") String reportingDate,
            @Field("accidentDate") String accidentDate,
            @Field("accidentAddress") String accidentAddress,
            @Field("disposalPerson") String disposalPerson,
            @Field("keyPerson") String keyPerson,
            @Field("reportingStaff") String reportingStaff,
            @Field("contactPhone") String contactPhone,
            @Field("mainDescription") String mainDescription,
            @Field("phoneftp") String phoneftp
    );

    @POST("rq/push/getDnGasEvents")
    @FormUrlEncoded
    Observable<SimpleHttpResult<List<EmergencyModel>>> getGasAccident(@Field("state") String state);

    @POST("rq/push/getDangerxian")
    @FormUrlEncoded
    Observable<SimpleHttpResult<EmergencyItem>> getAccidentDetail(@Field("shiguid") String state);
}

File: app/src/main/java/com/bkjcb/rqapplication/emergency/retrofit/EmergencyService.kt
package com.bkjcb.rqapplication.emergency.retrofit

import com.bkjcb.rqapplication.model.HttpResult
import io.reactivex.Observable
import retrofit2.http.Field
import retrofit2.http.FormUrlEncoded
import retrofit2.http.POST

/**
 * Created by DengShuai on 2019/10/31.
 * Description :
 */
interface EmergencyService {
    @POST("/rq/push/SaveSiteInfo")
    @FormUrlEncoded
    fun submit(
            @Field("userId") userId: String?,
            @Field("reportingUnit") reportingUnit: String?,
            @Field("qushu") qushu: String?,
            @Field("remark") remark: String?,
            @Field("reportingDate") reportingDate: String?,
            @Field("accidentDate") accidentDate: String?,
            @Field("accidentAddress") accidentAddress: String?,
            @Field("disposalPerson") disposalPerson: String?,
            @Field("keyPerson") keyPerson: String?,
            @Field("reportingStaff") reportingStaff: String?,
            @Field("contactPhone") contactPhone: String?,
            @Field("mainDescription") mainDescription: String?,
            @Field("phoneftp") phoneftp: String?
    ): Observable<HttpResult>
}